### ISSUE TITLE
navPlace Extension

### ISSUE THREAD

**Issue by glenrobson** [IIIF/trc#72]
## Links

 * Original Issue: https://github.com/IIIF/api/issues/1246
 * Pull Request: https://github.com/IIIF/api/pull/2016
 * Preview: https://preview.iiif.io/api/navplace_extension/api/extension/navplace/

## Background and Summary

The `navPlace` property has been discussed for a number of years but has been held back due to a lack of use cases and worries on the scope of the `navPlace` property. With the formation of the [Maps TSG](https://iiif.io/community/groups/maps-tsg/) this group has been able to discuss in detail the requirements for `navPlace` and are now in a position to submit this extension to the Presentation API to support linking a IIIF Manifest or Canvas to a geographical place. 

Use case that the navPlace will support include:

 * Linking a IIIF image of a map to the extent that it covers
 * Linking pages of an atlas to the parts of the World they cover
 * Geotagging a photograph to a location

Note just as with [navDate](https://iiif.io/api/presentation/3.0/#navdate), navPlace is intended to enable a viewer to relate a IIIF item to a geographical place and it does not allow you to specify the relationship between the place and the item for example you can not say the place is where the item was created, published or covers. 

## Proposed Solution

We hope the TRC will approve this extension. We have a number of implementations and also a couple of cookbook recipes demonstrating the new features that will be shared with the TRC once this extension is approved.

**Comment by azaroth42** [comment]
Recommended changes:

1. Add the requirements for `navPlace` as a property, in the style of the Presentation 3 API documentation. 

For example:
> * A Collection _MAY_ have the `navPlace` property with at least one entry.
   Clients _MAY_ render `navPlace` on a Collection.

2. Remove or provide explanation of the `properties` block in the example of the spec, and how it interacts with the `metadata` and `label` of the resource that has the `navPlace` extension added to it. What is a "IIIF compatible label"?

**Comment by thehabes** [comment]
Thank you @azaroth42, these are helpful points.  I neglected the "property" language and didn't even notice.  I will hang on that `properties` block comment for a moment during the call and see what people think.  

In response, what I really meant for the value is "IIIF Presentation API 3 compatible label"  It directs the user to use the language map (b/c that's IIIF Presentation API 3 friendly) as opposed to a standard string value `label: "like this"`.  Section 3.3 gets into explaining more about `properties` right after the full Manifest example.  

Maybe it is best to keep the navPlace snippet as simple as possible and just remove that `label`.  It doesn't really help the snippet anyway, since section 2.2 doesn't talk about metadata or the `properties` object.  It may also be worth adding a line like "See section 3.3 for more about using metadata with `properties`".

**Comment by kirschbombe** [comment]
Introduction
- Could this also be used for a celestial or spherical coordinate system? Does it have to be "earthbound"?
- Where is says "In IIIF Presentation API 3 the property navDate allows for a calendar date to be associated with a resource in a client friendly format. However, places are extents in space, independent of time and what may or may not be present in that space." Would it make more sense to say something like, "While the navDate property allows a machine-readable date..., navPlace..."
- change "client friendly" to "machine-readable"?

Objectives and scope
- The first paragraph covers objectives of IIIF, but it seems like it should instead focus on the objectives of navPlace

**Comment by thehabes** [comment]
Hi @kirschbombe!

Introduction
- It depends on how you look at it.  WGS84 is made up of a reference ellipsoid, a standard coordinate system, altitude data, and a geoid.  Nothing about that says it has to be applied to planet Earth.  However, the specific values it has that make up that data are geared for Earth.  The ellipsoid and geoid used are from Earth's center of mass, shape and specific spherical harmonics.  For other spheres, you should use values that offer the best precision for the sphere.  WGS84 was designed for Earth and is rarely used for other surfaces without adjustments.

- I don't want to the relationship between them to be seen as a corollary.  `navDate` is a good parallel as it shares similar technical and UX aspects, but we don't have `navPlace` "because of" `navDate`.  I want them to be though about explicitly separately.  Perhaps there is a clearer way to do this.
- The presi 3 spec on `navDate` says "A date that clients may use for navigation purposes..." which is why I went with "client friendly".  It does not use "machine-readable" because it will be "GeoJSON-LD" and no other format.  Look at `seeAlso`, for example.

Objectives and Scope
- Hmm good point.  Maybe flip the paragaphs in this section, and have the first paragraph trail into how navPlace fits the objectives of IIIF, or something like that.

Terminology
- The IIIF Presentation API 3 spec says "The terms array, JSON object, number, string, and boolean in this document are to be interpreted as defined by the Javascript Object Notation (JSON) specification."  So it is still a valid place to send peoplefor details on their meaning (is that cheating?) :)
- Section 2.1 does this, I think.

Position
- Yes.  See 2.1.4 and 2.1.5.  Those are straight out of the GeoJSON spec.

- They are for 2 different purposes.  2.1.1 shows you how to make the `@context` property for the web resource.  3.1 says what they are and why you need them.  See section 4.6 of presi 3, which also notes it twice (differently, but still twice).  So I would say yes, have both, but could be convinced otherwise.

**Comment by azaroth42** [comment]
Are there implementations for Mirador, UV or other "regular" IIIF client?

**Comment by glenrobson** [comment]
Not for navPlace yet, although there are examples using different ways of encoding geo coordinates in manifests before something like navPlace was available. See this example from UCD that uses a geo service:

https://digital.ucd.ie/view/ucdlib:256179

**Comment by azaroth42** [comment]
Explanation of my 😕  reaction -- not against the spec approach or principle, but think it needs a bit more work before publication.

**Comment by tomcrane** [comment]
Same as Rob I think -  the intent is to attach locations to IIIF resources by borrowing GeoJSON as a useful existing spec, rather than specifically to get GeoJSON objects into a GeoJSON aware environment. So better to disallow `properties` - it's the viewer's job to construct that if it wants to push the GeoJSON into a GeoJSON-aware rendering environment.

**Comment by kirschbombe** [comment]
Agree with @azaroth42, especially areas that could use more clarification.

**Comment by azaroth42** [comment]
One more ... 
> a single or multiple geographic areas 

Can the value of `navPlace` ever be an array of JSON objects? or is the 'multiple' via a FeatureCollection, and the value MUST be a JSON object?

**Comment by thehabes** [comment]
This is great, we needed to bring it outside of the IIIF Maps group to get to these.  The perspectives really help highlight what needs to be called out for other professionals trying to understand what this is.  From today, I have a much better idea on how to frame details around using metadata with this pattern.  A bit more about what the client is responsible for and what their options are for metadata would go a long way in helping to describe the scope.

**Comment by thehabes** [comment]
@azaroth42 the value cannot be an array of JSON objects.  It is always a single object, which is a Feature Collection, which can contain as many Features as you would like (until you break the internet).

**Comment by mixterj** [comment]
I am concerned about the lack of support for non-earth geo-locations. In a quick search, I was able to find an example of the [moon landing in 1969](https://researchworks.oclc.org/iiif-explorer/manifest?url=https%3A%2F%2Fcdm16694.contentdm.oclc.org%2Fdigital%2Fiiif-info%2Fp16694coll20%2F5799%2Fmanifest.json) 

I would like to add geo-coordinates to this image, how would I? To clarify, I very much like the GeoJson approach but I just want clarification about non-earth coordinates.

For reference Wikidata does account for non-earth coordinates in a rather nice way - [Mare Tranquillitatis](https://www.wikidata.org/wiki/Q320936)

**Comment by kirschbombe** [comment]
I'm also wondering if it would be useful at this stage to build in a way to give each Feature object a "motivation" of sorts. The issue mentions that the extension does not allow one to specify the relationship between the place and the item, but I think this can be an important distinction to make (we have different Place metadata fields in more complex schemas that allow us to specify that relationship (place of publication, subject, geographic coverage, etc.), especially where there might be several relationships at play in a single manifest or Feature Collection.

**Comment by thehabes** [comment]
@mixterj To show that photograph located to a specific point on the moon, you need

- A reference ellipsoid, a standard coordinate system, altitude data, and a geoid
- You need a viewer that shows the projection of that system on a 2D plane, hopefully with a basemap of the surface of the moon.
- The data to mark that location in the system on the viewer over the correct basemap project and connect the metadata to a UI for human readability

How would you suggest supplying all of that in a IIIF resource, staying Linked Data compatible and never breaking the schema?  Admittedly, I looked into it and cannot.  Specs are missing.  However, I found a way to do it for planet earth without having to write a coordinate system specification.

navPlace is meant to leverage available schemas, specifications and vocabularies that are widely used and already exist.  If you know of ones for  the moon, then you can make an extension for supplying moon related geospatial information.  If you can find a way to do it for any ellipsoid, then IIIF awaits your coordinate reference system extension :)

**Comment by azaroth42** [comment]
>  It is always a single object, which is a Feature Collection,

Then the definition of the value is MUST not SHOULD :)

There also should be more information, and an example, of the referenced rather than embedded pattern.

I _think_ what is meant is:

```
 "navPlace": {"id": "http://.../", "type": "FeatureCollection"}
```

with no `features` property?

Also, given the `Feature` example has an `id` ... it would be good to clarify if a Feature can be a reference or if they must be embedded.

**Comment by azaroth42** [comment]
Changing my vote to :-1: given the substantial nature of many of the above requested changes -- seems like they should be reviewed again by TRC.

**Comment by kirschbombe** [comment]
To accommodate other coordinate systems, could we for now add a property that specifies the system? So in the current case, it would be GeoJSON or WGS84. This would leave room to later use navPlace for other systems instead of creating new properties (like navPlaceMars :))

**Comment by thehabes** [comment]
@kirschbombe to give features a "motivation", you would have to extend the GeoJSON spec to include "motivation" and so not only would you need an extension for navPlace, but also an extension for GeoJSON. You can use GeoJSON in Web Annotation, where you have more control over motivation and purpose already.  

If you put the setting for coordinate systems it into `properties` then you need the linked data context to back it up.  Again, navPlace wants to use existing specs and standards, we are avoiding having to write any coordinate system specification of our own.

**Comment by zimeon** [comment]
I'm also :-1: at the moment based on the need for additional TRC discussion. Certainly want to see a `navPlace` extension agreed though. Thoughts:
* Need to understand whether to use `properties` or not. One consideration is whether the specification of multiple features in a FeatureCollection is intended to support independent features that might want individual labels (hence cannot inherit from IIIF description) or not (in which case `properties: {}` seems reasonable)
* I think GeoJSON for earthbound coordinates is the 99% use case so any solution that generalizes SHOULD NOT overly complicate the simple case. The use of scoped contexts relies upon `navPlace` to bring in the GeoJSON-LD context for contents so I think one would likely need at add another level of nesting to be able to say GeoJSON vs. other coordinate/representation types.

**Comment by glenrobson** [comment]
> I'm also wondering if it would be useful at this stage to build in a way to give each Feature object a "motivation" of sorts. The issue mentions that the extension does not allow one to specify the relationship between the place and the item, but I think this can be an important distinction to make (we have different Place metadata fields in more complex schemas that allow us to specify that relationship (place of publication, subject, geographic coverage, etc.). This would be especially helpful where there might be several relationships at play in a single manifest or Feature Collection.

Just to note that `navDate`  doesn't specify the date it is representing (e.g. publication or creation date etc.) so I think navPlace should be as vague. 

It should be noted though that `navDate` can only have one value so there is less need for a motivation.

**Comment by thehabes** [comment]
`it would be nice to be able to explicitly state the relationship between the IIIF resource and the place`
This is not possible in GeoJSON-LD.  Since we are using GeoJSON-LD, and not some derivative or extended version, you would have to use something IIIF Presentation API 3 has to do this.

There is a geo recipe that does get into this, published at https://iiif.io/api/cookbook/recipe/0139-geolocate-canvas-fragment/.  It uses motivation "tagging" because there is no published ontology or linked data extension for geographic motivations on Web Annotations.  You can write one though!  Bert is doing this for georeferencing, and you can see what work is involved at this repo: https://github.com/thehabes/GeoreferencePrototype.  You can also look at an example of writing a GeoJSON extension spec at https://github.com/kgeographer/geojson-t.  Remember the key phrase here "Using available web standards".  For what you want to do with this, a lot more extending is involved to make it Linked Data compatible and IIIF friendly. 

You can see how the OGC community handles some off this https://www.ogc.org/docs/is.  You will realize quickly how out of scope this becomes.  Right now, we just want to get in the door.  How do you connect geographic coordinates with IIIF resources in a standards compliant manner.  Attaching specific motivations and purposes to geospatial information are each their own individual challenge, as each will require some specific technical aspect that must be described.

**Comment by thehabes** [comment]
Certainly, `navPlace` is meant to spark these thoughts.  But it can't be the end all extension for geospatial practices.  Those practices are not all well described and need specifications of their own first before we can approach intertwining them with IIIF.  For now, navPlace can do what the web has standardized around: get shapes on a map and connect some metadata.  Essentially, everything else is under development with individual, differing implementations.

**Comment by tpendragon** [comment]
I want to see this here, but given the number of comments here I feel like we should see a review and another look. Things I'd particularly like to see are just some cleaning up of the language to more closely match the IIIF 3 format (it took me a while to figure out where I could attach a navPlace property) and I'd like to see this `properties` discussion worked out. I'm a little uncomfortable with part of the specification being "put whatever you want here" without some guidance as to why you might want some content there. I assume there's just external software that you want to interact with here? If the `properties` property just wasn't there at all, what wouldn't work?

I'm not concerned about it being earth-bound - I even prefer it, getting this integration really nailed down with the 99% use cases seems better and it can be refined later. IIIF has a long history of that, and I think it's part of what makes it great.

Is there a chance we'll be able to see this functionality do something in a viewer, too? Or an experience of some kind?

**Comment by glenrobson** [comment]
### Issue 72 (navPlace Extension)
  +1: 3 [glenrobson jcreel thehabes]
   0: 9 [cubap emulatingkat jtweed julsraemy markpatton mbennett-uoe nfreire shuddles tomcrane]
  -1: 9 [awead azaroth42 kirschbombe ksclarke mcwhitaker mixterj tpendragon triplingual zimeon]
  Not TRC: 1 [mapninja]
  Ineligible: 0 []

### Result: 3 / 21 = 0.14
Issue is rejected
